src/common/OfflineNotice.js

```diff
@@ -24,19 +26,76 @@ const styles = createStyleSheet({
 
 type Props = $ReadOnly<{||}>;
 
+const useHasStayedTrueForMs = (value: boolean, duration: number) => {
+  const timeoutId = useRef<TimeoutID | null>(null);
+  const [result, setResult] = useState<boolean>(false);
+
+  useEffect(() => {
+    if (value) {
+      timeoutId.current = setTimeout(() => {
+        setResult(true);
+      }, duration);
+    } else {
+      clearTimeout(timeoutId.current);
+      timeoutId.current = null;
+      setResult(false);
+    }
+  }, [value, duration]);
```

Comment:
Hmm. Won't this clobber the old timeout? That is, if we have:
* Render with value true.
* Render again with value true, before that timeout completes.

then we'll forget about the old timeout and set a new one.

I guess the original timeout will still end up firing… but because `timeoutId.current` won't know about it anymore, it'll end up firing even if the value *then* changes to false.

Oh, and I guess this is a latent bug as it stands, because it only triggers if the *effect* gets rerun. Which only happens if `value` or `duration` changes, and currently we never change `duration`. Still, good to fix the latent bug in case we reuse this -- plus I suspect the code will be somewhat easier to reason through in a fixed version.

Reply:
Separately: we should clear this timeout if the component unmounts, so we don't have a leaked timeout trying to belatedly call `setResult`.

`useEffect` supports this -- have its callback return the cleanup function:
https://reactjs.org/docs/hooks-reference.html#cleaning-up-an-effect

Reply:
> Oh, and I guess this is a latent bug as it stands, because it only triggers if the _effect_ gets rerun.

Ah, yeah, I think I was sort of counting on this fact for the implementation. I think I should use `usePrevious(value)` in there somewhere; I'll try that.